modules/labs-react/layout/stories/Stack.stories.mdx

```diff
@@ -0,0 +1,154 @@
+direction to `row` as is consistent with the CSS Flexbox spec.
+
+### Setting Space
+
+The space between child elements is set with the `spacing` prop, which supports all space token and
+string values. `Stack` intentionally only sets space between elements and never outside. For
+example: a vertical stack of three elements would only have margin added to the top of the second
+and third child elements. Keeping space between elements makes building layouts more block-like and
+predictable.
+
+### Managing Child Elements
+
+#### Basic Usage
+
+`Stack` uses CSS pseudo-classes to apply margin to child elements. This makes it really simple to
+add child elements and get consistent spacing without much manual effort. Below is a basic example.
+Note that it does not add any space above or below the child elements. This helps your element fit
+cleanly within their containers.
+
+<ExampleCodeBlock code={BasicStack} />
+
+#### Using `shouldWrapChildren`
+
+Because of the increased specificity of `Stack`'s pseduo-classes, it may override the child element
+margin styles. In that case, there are two recommended options available: Use the
+`shouldWrapChildren` prop or using `Stack.Item` (which is discussed in the section below). The
+`shouldWrapChildren` prop will wrap each child in a `Stack.Item` component and apply the margin to
+it instead of the child element. Below is an example with even space between the first three
+elements, and custom space between the third and fourth elements.
+
+<ExampleCodeBlock code={ShouldWrapChildren} />
+
+#### Using `Stack.Item`
+
+Sometimes inserting wrapper elements can be problematic when you need access to those elements. In
+those situations, `shouldWrapChildren` is not the best choice, and you should opt for wrapping
+children in `Stack.Item`s. Below we're wrapping each of the three elements in the first row with
+`Stack.Item` and applying `flex={1}` to them. This would not be possible with `shouldWrapChildren`.
+
+<ExampleCodeBlock code={StackItems} />
+
+### Valid Children
+
+Because `Stack` applies styles to children, it requires its immediate children to be valid child
+elements. For example, text outside of an HTML tag would not render if it was an immediate `Stack`
+child. Here's an example snippet:
+
+```tsx
+const ValidChildrenExample = () => {
+  return (
+    <Stack flexDirection="row" spacing="s">
+      This text will not render. Don't do this.
+      <p>This text will render.</p>
+      <span>This text will also render.</span>
+    </Stack>
+  );
+};
+```
+
+### Examples
+
+You can nest vertical and horizontal `Stack`s to create grid-like layouts. Here are three horizontal
+`Stack`s nested within a vertical `Stack`. The fourth row is a single `Flex` component.
+
+<ExampleCodeBlock code={NestedStacks} />
+
+You can also use `Stack` to manage smaller layouts, such as within Cards.
+
+<ExampleCodeBlock code={StackCard} />
+
+## HStack
```

Comment:
I've continuously questioned the value vs. overhead of `HStack` vs. `Vstack`. Is there anything I'm missing that makes them overly valuable? They barely seem like a convenience IMO (adding an import and changing components to save specifying a prop).

Reply:
That's a fair critique. `HStack` and `VStack` are less about the initial implementation, and more about reducing friction with layouts as you adapt them over time. For me it's less about saving a prop specification and more about understandability, especially in the context of nested . Similar to `Text` vs `Heading`, `Body`, et al. Let's use this markup as an example:

```tsx
<Stack flexDirection="column" spacing="s">
  <Stack flexDirection="row" spacing="s">
    <Box flex={1}>One</Box>
    <Box flex={1}>Two</Box>
    <Box flex={1}>Three</Box>
  </Stack>
  <Stack flexDirection="row" spacing="s">
    <Box flex={1}>Four</Box>
    <Box flex={1}>Five</Box>
  </Stack>
  <Stack flexDirection="row" spacing="s">
    <Box flex={1}>Six</Box>
    <Box flex={1}>Seven</Box>
    <Box flex={1}>Eight</Box>
  </Stack>
</Stack>
```

While I can interpret the stack's direction by reading the props, it takes a bit more effort than this:

```tsx
<VStack spacing="s">
  <HStack spacing="s">
    <Box flex={1}>One</Box>
    <Box flex={1}>Two</Box>
    <Box flex={1}>Three</Box>
  </HStack>
  <HStack spacing="s">
    <Box flex={1}>Four</Box>
    <Box flex={1}>Five</Box>
  </HStack>
  <HStack spacing="s">
    <Box flex={1}>Six</Box>
    <Box flex={1}>Seven</Box>
    <Box flex={1}>Eight</Box>
  </HStack>
</VStack>
```

If I'm a developer trying to figure out which `Stack` needs to be adjusted for the layout or where I need to put a new item, I'll opt for `HStack` and `VStack` every time.

Reply:
That said, I didn't explain any of that in the docs, and I should. 😄

Reply:
**Re: Overhead** 

I agree that overhead is an important consideration for us. I would argue that the cost of `HStack` and `VStack` is quite low. Each component is only 11 lines of code (14 if you include the types).

```tsx
export const HStack = createComponent('div')({
  displayName: 'HStack',
  Component: ({children, flexDirection = 'row', ...elemProps}: HStackProps, ref, Element) => {
    return (
      <Stack as={Element} ref={ref} flexDirection={flexDirection} {...elemProps}>{children}</Stack>
    );
  },
    subComponents: {
      Item: StackItem,
  },
});
```

Reply:
Fair enough - you sold me 😛